common/src/worldgen.rs

```diff
@@ -167,83 +169,87 @@ impl ChunkParams {
         self.chunk
     }
 
-    fn generate_terrain(&self, center: na::Vector3<f64>) -> Material {
-        let cube_coords = center * 0.5;
-
-        let rain = trilerp(&self.env.rainfalls, cube_coords);
-        let temp = trilerp(&self.env.temperatures, cube_coords);
-        let slope = trilerp(&self.env.slopeinesses, cube_coords);
-        let flat = trilerp(&self.env.flatness, cube_coords);
-
-        // block is a real number, threshold is in (0, 0.2) and biased towards 0
-        // This causes the level of terrain bumpiness to vary over space.
-        let block = trilerp(&self.env.blockinesses, cube_coords);
-        let threshold = 2.0f64.powf(block) / (4.0 + 2.0f64.powf(block)) * 0.2;
-        let elev_raw = trilerp(&self.env.max_elevations, cube_coords);
-        let terracing_scale = 5.0; // This is not wavelength in number of blocks
-        let elev_floor = (elev_raw / terracing_scale).floor();
-        let elev_rem = elev_raw / terracing_scale - elev_floor;
-        let elev = terracing_scale * elev_floor + serp(0.0, terracing_scale, elev_rem, threshold);
-
-        let mut voxel_mat;
-        let max_e;
-
+    fn choose_base_material(rain: f64, temp: f64) -> Material {
```

Comment:
This doesn't interact with `ChunkParams` in any way, so it probably makes more sense as a module-level function rather than an associated function.